broker/core/src/bbdo/stream.cc

```diff
@@ -1144,6 +1153,7 @@ bool stream::read(std::shared_ptr<io::data>& d, time_t deadline) {
  *  @return Respect io::stream::read()'s return value.
  */
 bool stream::_read_any(std::shared_ptr<io::data>& d, time_t deadline) {
+  READ_FROM_QUEUE
```

Comment:
I think we can split this code more explicitly.
It looks strange to call READ_FROM_QUEUE everywhere, I have the feeling this function can be splitted in two parts.